openedx/core/djangolib/model_mixins.py

```diff
@@ -12,3 +12,25 @@ def __init__(self, *args, **kwargs):  # pylint: disable=unused-argument
         Override to kill usage of this model.
         """
         raise TypeError("This model has been deprecated and should not be used.")
+
+
+class DeletableByUserEmail(object):
+    """
+    This mixin allows inheriting models to delete users by their email field.
+    """
+
+    @classmethod
+    def retire_user(cls, user_email):
```

Comment:
A better name would be ‘delete_record’

Reply:
@schenedx, this was a naming convention @iloveagent57 agreed to. I'm fine with the change.  @iloveagent57, your thoughts?

Reply:
The function is to delete the db row, not really retire the user email. I think the name of the function should reflect accurately what the function does.